tutorials/plot_background_morph.py

```diff
@@ -0,0 +1,875 @@
+complex operations than just shifting brains relative to each other.
+Nevertheless, this can be a first step of a successful subject to subject
+transformation.
+
+We can understand mapping as a mathematical function :math:`f`. This function
+can be seen as a description on how to move from one point in n-dimensional
+space to another. Like with a real maps, when attempting to reach a new real
+world location, we try to derive a set of actions (in the example directional
+decisions to reach the new location), to move from one point to another. This
+set of actions can also be described as a function of the current position
+that computes the new location.
+
+In general morphing operations can be split into two different kinds of
+transformation: linear and non-linear morphs or mappings:
+
+A mapping is linear if it satisfies the following two conditions:
+
+.. math::    f(u + v) = f(u) + f(v)\ ,
+.. math::    f(cu) = cf(u)\ ,
+
+where :math:`u` and :math:`v` are from the same vector space and :math:`c` can
+be any scalar. This means that any linear transform can be described as a
+mixture of additive and multiplicative operations and hence is often
+represented in terms of a transformation matrix:
+
+.. math::
+
+    x^{(B)}=
+    \begin{bmatrix}
+    x\prime \\
+    y\prime \\
+    z\prime \\
+    1 \end{bmatrix} =
+    M^{(AB)}x^{(A)}=
+    \begin{bmatrix}
+    1 & 0 & 0 & 0 \\
+    0 & 1 & 0 & 0 \\
+    0 & 0 & 1 & 0 \\
+    0 & 0 & 0 & 1
+    \end{bmatrix}
+    \begin{bmatrix}
+    x \\
+    y \\
+    z \\
+    1
+    \end{bmatrix}\ ,
+
+where :math:`x^{(A)}` is the data that is morphed (subject A) and
+:math:`x^{(B)}` the morphed data (data of subject A transformed into the space
+of subject B). :math:`M^{(AB)}` represents the transformation matrix. In that
+case the identity matrix does not cause any change such that
+:math:`x^{(A)} = x^{(B)}`\ .
+
+The simplest example on how transformation matrices work, would be scaling. If
+we replace the :math:`1` in the second column of :math:`M^{(AB)}`\ , only
+:math:`y` will be multiplied with this value. Changing this value therefore
+scales all values :math:`y`\ .
+
+Other operations like rotation, shearing, reflection, etc. will not be
+explained here, however the general idea is the same as for scaling.
+Translation is a little bit of a special case and furthermore the reason, why
+an additional :math:`1` was added to :math:`\vec{d}`\ . It is mathematically
+impossible to translate data represented in in N dimensions within
+in :math:`R^N`. Thus, an additional dimension is added such that the space
+becomes :math:`R^{N+1}`. Now we can exploit the fact, that shearing in
+:math:`R^{N+1}` causes a translation in :math:`R^N`. By keeping the added
+dimension constant at :math:`1` our data that originally was in :math:`R^N`,
+will be sheared in :math:`R^{N+1}` - not affecting the additional dimension
+:math:`N+1`, but causing a translation in the respective other N dimensions.
```

Comment:
The last two paragraphs feel like overly verbose to me. In my opinion, we should simply point to the wikipedia page for affine transformations: https://en.wikipedia.org/wiki/Transformation_matrix#Affine_transformations

Reply:
... ok :(

But do you think it's obvious to people who never thought about transformation matrices, why translation actually requires an additional dimension (I just really like the way of expressing this as shearing in higher dimension)? Also: wikipedia is also not so nice and clear about that ;)

removing?